R/get_oaproc.R

```diff
@@ -0,0 +1,289 @@
+                                           value = p[2]),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = p[1]),
+                                      list(id = "raindroptrace",
+                                           type = "text/plain",
+                                           value = ifelse(raindrop,
+                                                          "true", "false")),
+                                      list(id = "direction",
+                                           type = "text/plain",
+                                           value = direction))),
+                   pretty = TRUE, auto_unbox = TRUE)
+}
+
+make_json_input_split <- function(p, upstream = TRUE) {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
+                                           type = "text/plain",
+                                           value = p[2]),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = p[1]),
+                                      list(id = "upstream",
+                                           type = "text/plain",
+                                           value = ifelse(upstream,
+                                                          "true", "false")))),
+                   pretty = TRUE, auto_unbox = TRUE)
+
+}
+
+make_json_input_xspt <- function(p, w, n) {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
+                                           type = "text/plain",
+                                           value = p[2]),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = p[1]),
+                                      list(id = "width",
+                                           type = "text/plain",
+                                           value = w),
+                                      list(id = "numpts",
+                                           type = "text/plain",
+                                           value = n))),
+                   pretty = TRUE, auto_unbox = TRUE)
+}
+
+
+make_json_input_xspts <- function(p1, p2, n) {
+
+  jsonlite::toJSON(list(inputs = list(list(id = "lat",
+                                           type = "text/plain",
+                                           value = c(p1[2], p2[2])),
+                                      list(id = "lon",
+                                           type = "text/plain",
+                                           value = c(p1[1], p2[1])),
+                                      list(id = "3dep_res",
+                                           type = "text/plain",
+                                           value = "1"),
```

Comment:
What are other values of this?

Reply:
Sorry about the lack of documentation - I'll get to that asap.  Its intended to be one of the three native resolutions of 3DEP data 1, 3, 5 10, 30, 60.  This is a helpful question.  To be honest I'm not sure if it can be any whole or integer number between 1-60, and I'm not sure what would happen if say your cross-section had a width of 10m and you chose a resolution of 30m - so I'll set up some tests around that and be sure to trap for bad input. It's documented pretty well in the packages CLI but should also look into how to better document the process metadata.

Reply:
Got it. Makes sense.